core-plugins/src/test/java/co/cask/hydrator/plugin/batch/source/FileBatchSourceTest.java

```diff
@@ -463,4 +475,171 @@ public void testRecursiveTimeFilterRegex() throws Exception {
     Assert.assertTrue(outputValue.contains("Hello,World"));
     Assert.assertTrue(outputValue.contains("CDAP,Platform"));
   }
+
+  @Test
+  public void testFileBatchInputFormatText() throws Exception {
+    String filePath = "file:///tmp/test/test.txt";
+    String testData = "String for testing purposes";
+
+    Path textFile = new Path(filePath);
+    Configuration conf = new Configuration();
+    FileSystem fs = FileSystem.get(conf);
+    FSDataOutputStream writeData = fs.create(textFile);
+    writeData.write(testData.getBytes());
```

Comment:
don't need to write through the Hadoop FileSystem abstraction, just do it through normal java File.